pyfr/plugins/solnwriter.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+
+from pyfr.h5writer import H5Writer
+from pyfr.inifile import Inifile
+from pyfr.plugins.base import BasePlugin
+
+
+class SolnWriterPlugin(BasePlugin):
+    name = 'solnwriter'
+    systems = ['*']
+
+    def __init__(self, intg, cfgsect, suffix=None):
+        super().__init__(intg, cfgsect, suffix)
+
+        basedir = self.cfg.getpath(cfgsect, 'basedir', '.')
+        basename = self.cfg.get(cfgsect, 'basename', raw=True)
+        serial_h5 = self.cfg.getbool(cfgsect, 'serial-h5', False)
+
+        self._writer = H5Writer(intg, basedir, basename, 'soln', serial_h5)
+        self.tsteps = self.cfg.getfloat(self.cfgsect, 'tsteps')
```

Comment:
Like you, I am not in favour of 'dt' variable by itself because name clashes with integration dt.
Along with call_plugin_ I guessed the name is explicitly clear. 

Besides nsteps rhymes with tsteps, make it more obvious (-- poetic logic?)

If we prefer, we could name call_plugin_dt to 
call_plugins_at_tstep /
call_plugin_tstep /even 
plugin_tstep ?